rerun_py/rerun_sdk/rerun/_rerun2/archetypes/__init__.py

```diff
@@ -2,7 +2,8 @@
 
 from __future__ import annotations
 
-__all__ = ["AffixFuzzer1", "Points2D"]
+__all__ = ["Archetype", "AffixFuzzer1", "Points2D"]
 
+from ._base import Archetype
```

Comment:
I'm not a big fan of having the abstract super class (or whatever it's called) be defined in this package with the rest of the actual archetypes...
In particular I can very easily imagine a situation where someone wants an archetype called `Archetype` :grimacing: 

Can we put those abstract classes at the very top maybe? `_rerun2.Archetype`, `_rerun2.Component`, etc

Reply:
Moving them all to `_rerun2/_baseclasses.py`.

Note that in the next PR:
- there will be many more such base classes 
- most of them will have longer names...
- ...except `Archetype`, which is user-facing (that's what `log_any()` accepts)